### PR TITLE
Shayan/Add prop types to API documentation

### PR DESCRIPTION
**Pull request by shayan-binary**

### CODE REVIEW

File: src/components/AssetInformation.jsx [hunk 1]
```diff
@@ -21,6 +22,24 @@ const AssetInformation = ({
     </div>
 );
 
+AssetInformation.propTypes = {
+    price       : PropTypes.string,
```
Comment (nazaninreihani) [inline src/components/AssetInformation.jsx]: there are lots of extra space here and in `AssetInformation.defaultProps `, is that needed?

File: src/components/Barrier.jsx [hunk 2]
```diff
@@ -27,6 +28,28 @@ const Barrier = ({
     </div>
 ));
 
+Barrier.propTypes = {
```
Comment (nazaninreihani) [inline src/components/Barrier.jsx]: There are lots of space between property and value in all files, I'm not sure why we need them

File: src/components/ChartTypes.jsx [hunk 3]
```diff
@@ -64,11 +65,34 @@ const ChartTypes = ({
                         )}
                     </ChartTypeList>
                 </div>
-            </ChartTypeMenu.Body>
+            </ChartTypeMenu.Body>w
```
Comment (nazaninreihani) [inline src/components/ChartTypes.jsx]: what is `w`?

**Comment by morteza-binary** [comment]
@shayanbinary would you please solve the conflicts?

File: src/components/AnimatedPrice.jsx [hunk 4]
```diff
@@ -14,4 +15,19 @@ const AnimatedPrice = ({
     );
 };
 
+AnimatedPrice.propTypes = {
+    isIncrease  : PropTypes.bool,
```
Comment (morteza-binary) [inline src/components/AnimatedPrice.jsx]: @shayanbinary Also here, there are lots of white spaces before the colon.

File: src/components/Dialog.jsx [hunk 5]
```diff
@@ -15,4 +16,18 @@ const Dialog = ({
...
 
+Dialog.propTypes = {
+    open                : PropTypes.bool,
+    children            : PropTypes.any,
+    onContainerClick    : PropTypes.func,
+    className           : PropTypes.string,
+};
+
+Dialog.defaultProps = {
+    open                : PropTypes.bool,
```
Comment (morteza-binary) [inline src/components/Dialog.jsx]: @shayanbinary `PropTypes.bool` is not default value for a propery.

File: src/components/Chart.jsx [hunk 6]
```diff
@@ -145,6 +146,90 @@ class Chart extends Component {
...
+    scrollToEpoch                   : PropTypes.number,
+    scrollToEpochOffset             : PropTypes.number,
+    zoom                            : PropTypes.number,
+    clearChart                      : PropTypes.bool,
+    onExportLayout                  : PropTypes.func,
+    importedLayout                  : PropTypes.func,
+};
+
+Chart.defaultProps = {
+    id                              : undefined,
```
Comment (morteza-binary) [inline src/components/Chart.jsx]: @shayanbinary it doesn't need to set a default value when is `null`, `undefined`, `''` and `false`

File: src/components/PriceLine.jsx [hunk 7]
```diff
@@ -41,4 +42,36 @@ class PriceLine extends Component {
...
+    visible         : PropTypes.bool,
+    zIndex          : PropTypes.number,
+    uncentered      : PropTypes.bool,
+    offScreen       : PropTypes.bool,
+    lineStyle       : PropTypes.string,
+    color           : PropTypes.string,
+    foregroundColor : PropTypes.string,
+};
+
+PriceLine.defaultProps = {
```
Comment (morteza-binary) [inline src/components/PriceLine.jsx]: @shayanbinary totally, there isn't any need for the default values for this component.

File: src/components/SearchInput.jsx [hunk 8]
```diff
@@ -34,6 +37,18 @@ class SearchInput extends Component {
...
 }
 
+SearchInput.propTypes = {
+    placeholder : PropTypes.string,
+    value       : PropTypes.string,
+    searchInput : PropTypes.object,
+};
+
+SearchInput.defaultProps = {
+    placeholder : '',
```
Comment (morteza-binary) [inline src/components/SearchInput.jsx]: @shayanbinary ditto

File: src/components/Dialog.jsx (hunk 5 above)
Comment (shayanbinary) [reply src/components/Dialog.jsx]: oops!

File: src/components/Barrier.jsx [hunk 9]
```diff
@@ -1,11 +1,12 @@
-import React from 'react';
-import { connect } from '../store/Connect';
+import PropTypes    from 'prop-types';
```
[the remark this answers is not in the document]
Comment (nazaninreihani) [reply src/components/Barrier.jsx]: @morteza-binary what do you think?

File: src/components/ChartTable.jsx [hunk 10]
```diff
@@ -124,6 +125,22 @@ const ChartTable = ({
     );
 };
 
+ChartTable.propTypes = {
+    isMobile: PropTypes.bool,
+    tableData: PropTypes.array,
+    Dialog: PropTypes.any.isRequired,
+    open: PropTypes.bool,
```
[the remark this answers is not in the document]
Comment (shayanbinary) [reply src/components/ChartTable.jsx]: `open`? I just double check it, it's a `boolean` prop

File: src/components/AssetInformation.jsx [hunk 11]
```diff
@@ -21,6 +22,23 @@ const AssetInformation = ({
...
+    open: PropTypes.string,
+    close: PropTypes.string,
+    high: PropTypes.string,
+    low: PropTypes.string,
+    visible: PropTypes.bool,
+};
+
+AssetInformation.defaultProps = {
+    price: '0',
+    open: 0,
```
[the remark this answers is not in the document]
Comment (shayanbinary) [reply src/components/AssetInformation.jsx]: well, I think we can put an integer into a string variable, anyway I fixed this.

File: src/components/Barrier.jsx (hunk 9 above)
[the remark this answers is not in the document]
Comment (morteza-binary) [reply src/components/Barrier.jsx]: I couldn't agree more.

File: src/components/ChartTable.jsx (hunk 10 above)
[the remark this answers is not in the document]
Comment (morteza-binary) [reply src/components/ChartTable.jsx]: It should be bool. Unfortunately, There isn't any name convention in this project, I should follow the code style in this project too.

[the remark this answers is not in the document]
Comment (nazaninreihani) [reply src/components/ChartTable.jsx]: oops, my bad, you're right

**Comment by shayan-binary-2** [comment]
@morteza-binary  Please close this PR. at last, we decide to stop the process for that moment.

**Pull request closed without merging**